crypto_kem/mceliece348864/clean/benes.c

```diff
@@ -0,0 +1,139 @@
+/*
+  This file is for Benes network related functions
+*/
+
+#include "benes.h"
+
+#include "params.h"
+#include "transpose.h"
+#include "util.h"
+
+/* one layer of the benes network */
+static void layer(uint64_t *data, uint64_t *bits, int lgs) {
+    int i, j, s;
+
+    uint64_t d;
+
+    s = 1 << lgs;
+
+    for (i = 0; i < 64; i += s * 2) {
+        for (j = i; j < i + s; j++) {
+
+            d = (data[j + 0] ^ data[j + s]);
+            d &= (*bits++);
+            data[j + 0] ^= d;
+            data[j + s] ^= d;
+        }
+    }
+}
+
+/* input: r, sequence of bits to be permuted */
+/*        bits, condition bits of the Benes network */
+/*        rev, 0 for normal application; !0 for inverse */
+/* output: r, permuted bits */
+void PQCLEAN_MCELIECE348864_CLEAN_apply_benes(unsigned char *r, const unsigned char *bits, int rev) {
```

Comment:
fixed size integers?

Reply:
We don't require that outside of `api.h`.